doc/howto/network_bridge_firewalld.md

```diff
@@ -1,52 +1,109 @@
+Nftables adds the ability to separate rules into namespaces, which allows multiple systems to add rules that won't affect each other.
+
+If your system supports and uses nftables, LXD detects this and switches to nftables mode.
+In this mode, LXD adds its rules into the nftables, using its own nftables namespace.
+As a result, there are usually no conflicts with other applications.
+
+If your system uses xtables, on the other hand, you might see rule conflicts.
+In this case, follow the instructions below to configure LXD's rules directly in xtables instead of having LXD add them automatically.
+
+### Disable LXD's xtables rules
+
+Run the following commands to prevent LXD from setting xtables rules for a specific network bridge (for example, `lxdbr0`):
+
+    lxc network set <network_bridge> ipv4.nat false
+    lxc network set <network_bridge> ipv6.nat false
+    lxc network set <network_bridge> ipv6.firewall false
+    lxc network set <network_bridge> ipv4.firewall false
+
+### Firewalld: Add LXD xtables rules
+
+To configure the LXD xtables rules in firewalld, you must add input, output, forward and postrouting rules for the network bridge (for example, `lxdbr0`) and the associated IP range (for example, `192.0.2.0/24`):
+
+    firewall-cmd --permanent --direct --add-rule ipv4 filter INPUT 0 -i <network_bridge> -s <IP_range> -m comment --comment "generated by firewalld for LXD" -j ACCEPT
+    firewall-cmd --permanent --direct --add-rule ipv4 filter OUTPUT 0 -o <network_bridge> -d <IP_range> -m comment --comment "generated by firewalld for LXD" -j ACCEPT
+    firewall-cmd --permanent --direct --add-rule ipv4 filter FORWARD 0 -i <network_bridge> -s <IP_range> -m comment --comment "generated by firewalld for LXD" -j ACCEPT
+    firewall-cmd --permanent --direct --add-rule ipv4 nat POSTROUTING 0 -s <IP_range> ! -d <IP_range> -m comment --comment "generated by firewalld for LXD" -j MASQUERADE
+    firewall-cmd --reload
+
+Use the following command to check that the rules are taken into account by firewalld:
+
+    firewall-cmd --direct --get-all-rules
+
+<!-- Include start warning -->
+```{warning}
+The commands given above show a simple example configuration.
+Depending on your use case, you might need more advanced rules and the example configuration might inadvertently introduce a security risk.
 ```
+<!-- Include end warning -->
 
-Warning: what is exposed above is not a fool-proof approach and may end up inadvertently introducing a security risk.
+### UFW: Add LXD xtables rules
+
+To configure the LXD xtables rules in UFW, you must allow incoming, outgoing and routing traffic for the network bridge (for example, `lxdbr0`) and the associated IP range (for example, `192.0.2.0/24`):
+
+    ufw allow in on <network_bridge> from <IP_range>
+    ufw allow out on <network_bridge> to <IP_range>
+    ufw route allow in on <network_bridge> out on <network_bridge> to <IP_range>
```

Comment:
These are plain guesses, and I have no idea how to "translate" the postrouting rule above. ;)

Reply:
But this is exactly what we already say in the section above (for "Allow traffic from/to the LXD bridge").
The idea was that that section describes what you have to do in any case if you're running a firewall, just to make sure the bridge is accessible.
The other section ("Prevent rule conflicts") describes what to do if you are seeing conflicts in the rules (namely move the rules that LXD has directly into the firewall). 

That's why I asked if the first section actually makes sense for UFW.

Reply:
It both makes sense and doesn't (in that it causes duplication - although I didn't know that until I'd spent some time researching ufw and found it doesn't apparently support setting up SNAT rules without adding manual iptables rules to a file).

Ultimately ufw's default block rules are the same (in terms of end result) as rule conflicts caused by ordering issues. 
So I think it would be easier to keep the first section, but expand it with additional explanation from the `Prevent rule conflicts` section, so the ordering would be:

Intro:
Linux firewalls are based on netfilter....

xtables vs nftables (from https://github.com/lxc/lxd/blob/8a77eb65cd9f92ee8b623a4b5b01a8665ba3b3ad/doc/howto/network_bridge_firewalld.md#xtables-vs-nftables)

Disable LXD's firewall rules (from https://github.com/lxc/lxd/blob/8a77eb65cd9f92ee8b623a4b5b01a8665ba3b3ad/doc/howto/network_bridge_firewalld.md#disable-lxds-firewall-rules) but without disabling NAT.

Allow traffic from/to the LXD bridge:
https://github.com/lxc/lxd/blob/8a77eb65cd9f92ee8b623a4b5b01a8665ba3b3ad/doc/howto/network_bridge_firewalld.md#allow-traffic-fromto-the-lxd-bridge

LXD and Docker (from https://github.com/lxc/lxd/blob/8a77eb65cd9f92ee8b623a4b5b01a8665ba3b3ad/doc/howto/network_bridge_firewalld.md#lxd-and-docker)


I've also found issues with the https://github.com/lxc/lxd/blob/8a77eb65cd9f92ee8b623a4b5b01a8665ba3b3ad/doc/howto/network_bridge_firewalld.md#firewalld-add-lxd-firewall-rules rules as they don't appear to work when using a system with nftables (even with the xtables nftables shim).